### PR TITLE
Add method to get and add a goal

### PR DESCRIPTION
**Pull request by Het7**
## Notion ticket link

[Add method to get and add a goal](https://www.notion.so/uwblueprintexecs/Add-method-to-get-and-add-a-goal-5c1d3c9da91743b99c580816d2cb96bc)

## Implementation description
* followed along branches PR

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/resources/goal_dto.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        self.id = id
+        self.goal = goal.upper()
+
+
+class CreateGoalDTO:
+    def __init__(self, goal):
+        error = self.validate(goal)
+        if error:
+            raise Exception(error)
+        self.goal = bragoalnch.upper()
```
Comment (jennchenn) [inline backend/python/app/resources/goal_dto.py:12]:
```suggestion
        self.goal = goal.upper()
```

File: backend/python/app/services/implementations/goal_service.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from ...models import db
+from ...models.goal import Goal
+from ...resources.goal_dto import GoalDTO
+from ..interfaces.goal_service import IGoalService
+
+
+class GoalService(IGoalService):
```
Comment (jennchenn) [inline backend/python/app/services/implementations/goal_service.py:7]: @thrvg made some changes to this sort of logic in #48 

for example, take a look at: https://github.com/uwblueprint/childrens-aid-society/blob/main/backend/python/app/services/implementations/transport_method_service.py

File: backend/python/app/services/implementations/goal_service.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+            if goal_entry:
+                return GoalDTO(goal_entry.id, goal_entry.goal)
+            else:
+                return self.add_new_goal(goal_upper)
+        except Exception as error:
+            self.logger.error(str(error))
+            raise error
+
+    def add_new_goal(self, goal):
```
Comment (jennchenn) [inline backend/python/app/services/implementations/goal_service.py:24]: this method should accept a `type` parameter as well

File: backend/python/app/resources/goal_dto.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class GoalDTO:
+    def __init__(self, id, goal):
+        self.id = id
+        self.goal = goal.upper()
+
+
+class CreateGoalDTO:
+    def __init__(self, goal):
```
Comment (jennchenn) [inline backend/python/app/resources/goal_dto.py:8]: you should add a `type` parameter here

File: backend/python/app/resources/goal_dto.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class GoalDTO:
+    def __init__(self, id, goal):
```
Comment (jennchenn) [inline backend/python/app/resources/goal_dto.py:2]: you should add a `type` parameter here

File: backend/python/tests/functional/test_goal_service.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import pytest
```
Comment (jennchenn) [inline backend/python/tests/functional/test_goal_service.py:1]: these tests don't seem to be passing atm - can you make the above changes, and add in a `type` to these default goals as well?

**Review by jennchenn — CHANGES_REQUESTED** [review]
requesting some changes!

for reference, please take a look at how the goal table is set up here: https://github.com/uwblueprint/childrens-aid-society/blob/main/backend/python/app/models/goal.py and also @thrvg's old PR linked in the comments below!

File: backend/python/app/services/implementations/goal_service.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+from ...models.goal import Goal
+from ...resources.goal_dto import GoalDTO
+from ..interfaces.goal_service import IGoalService
+
+
+class GoalService(IGoalService):
+    def __init__(self, logger):
+        self.logger = logger
+
+    def get_goal(self, goal):
```
Comment (jennchenn) [inline backend/python/app/services/implementations/goal_service.py:11]: do you mind making two separate get methods here: one for `get_short_term_goal` and `get_long_term_goal`

they would essentially look the same as your current get method, but with an added `type=` in the filter

File: backend/python/tests/functional/test_goal_service.py (hunk 6 above)
Comment (Het7) [reply backend/python/tests/functional/test_goal_service.py:1]: Yeah for sure! 
Would a goal with `type` look like this:
`{"SHORT_TERM": "Caregiver(s) encourage/allow"}`

Comment (jennchenn) [reply backend/python/tests/functional/test_goal_service.py:1]: if you take a look at the model, it should have a `concern` field and a `type` field

/take a look at the migration from Andrew's squash concerns table PR it should have some examples there you can use !!!

File: backend/python/app/services/implementations/goal_service.py (hunk 7 above)
Comment (Het7) [reply backend/python/app/services/implementations/goal_service.py]: Sounds good, I have made the changes!

**Review by jennchenn — APPROVED** [review]

**Pull request merged**